pkg/api/validation/validation.go

```diff
@@ -1237,6 +1261,23 @@ func ValidatePersistentVolume(pv *api.PersistentVolume) field.ErrorList {
 			allErrs = append(allErrs, validateScaleIOVolumeSource(pv.Spec.ScaleIO, specPath.Child("scaleIO"))...)
 		}
 	}
+	if pv.Spec.LocalStorage != nil {
+		if numVolumes > 0 {
+			allErrs = append(allErrs, field.Forbidden(specPath.Child("localStorage"), "may not specify more than 1 volume type"))
+		} else {
+			numVolumes++
+			allErrs = append(allErrs, validateLocalStorageVolumeSource(pv.Spec.LocalStorage, specPath.Child("localStorage"))...)
+			// StorageClass is required
+			if len(pv.Spec.StorageClassName) == 0 {
```

Comment:
What I actually want to verify is that the StorageClass exists and also has the topology-key set.  I'm not sure if we can do that at this point though.